app/src/main/java/com/github/sdp/tarjetakuna/ui/webapi/WebApiFragment.kt

```diff
@@ -43,20 +43,47 @@ class WebApiFragment : Fragment() {
 
         // set the buttons listeners for the webapi
         binding.apiCards.setOnClickListener {
+            hideKeyboard()
             binding.apiResults.text = getString(R.string.api_waiting_results)
             viewModel.getCards(this.requireContext())
         }
+
+        binding.apiCardsByName.setOnClickListener {
+            hideKeyboard()
+            binding.apiResults.text = getString(R.string.api_waiting_results)
+            viewModel.getCardsByName(this.requireContext(), binding.cardName.text.toString())
+        }
+
+        binding.apiCardsBySet.setOnClickListener {
+            hideKeyboard()
+            binding.apiResults.text = getString(R.string.api_waiting_results)
+            viewModel.getCardsBySet(this.requireContext(), binding.setIdForCards.text.toString())
+        }
+
+        binding.apiCardById.setOnClickListener {
+            hideKeyboard()
+            binding.apiResults.text = getString(R.string.api_waiting_results)
+            viewModel.getCardById(this.requireContext(), binding.cardId.text.toString())
+        }
+
         binding.apiSets.setOnClickListener {
+            hideKeyboard()
             binding.apiResults.text = getString(R.string.api_waiting_results)
             viewModel.getSets(this.requireContext())
         }
```

Comment:
Looks like those listeners have a lot of common code that could be factored into one reusable function